crates/autopilot/src/decoded_settlement.rs

```diff
@@ -66,41 +68,83 @@ pub struct DecodedTrade {
     pub fee_amount: U256,
     pub flags: TradeFlags,
     pub executed_amount: U256,
-    pub signature: Bytes<Vec<u8>>,
+    pub signature: Signature,
 }
 
-impl DecodedTrade {
-    /// Returns the signature of the order.
-    fn signature(&self) -> Result<Signature> {
-        Signature::from_bytes(self.flags.signing_scheme(), &self.signature.0)
+pub struct RawTrade {
+    sell_token_index: U256,
+    buy_token_index: U256,
+    receiver: Address,
+    sell_amount: U256,
+    buy_amount: U256,
+    valid_to: u32,
+    app_data: Bytes<[u8; 32]>,
+    fee_amount: U256,
+    flags: U256,
+    executed_amount: U256,
+    signature: Bytes<Vec<u8>>,
+}
```

Comment:
The intermediate type is useful for logging when the trade is not yet decoded, but we need something in the log already in case of an error. Printing a raw tuple would look useless there.

Reply:
This log

https://github.com/cowprotocol/services/blob/jit-orders-surplus-exempt/crates/autopilot/src/decoded_settlement.rs#L271